clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -193,11 +193,8 @@ public NetworkClientDelegate.PollResult poll(long currentTimeMs) {
         }
         pollTimer.update(currentTimeMs);
         if (pollTimer.isExpired() && !membershipManager.isLeavingGroup()) {
-            logger.warn("Consumer poll timeout has expired. This means the time between " +
```

Comment:
Can you actually leave this log untouched? On the one hand I kind of agree with this simplification, and logs are by no means a part of the public contract, but I know for a fact that some people have built observation tools and/or dashboards for things like rebalancing issues by searching for relevant log strings such as this one (I know because I built one myself a long time ago)

I don't feel super strongly about this so  I won't push back if you'd prefer to clean it up, but imo it doesn't hurt to leave the log here as well

Also: in some extreme cases, eg an infinite loop in a user's processing logic, the consumer might never return to call `poll` at all. In less extreme cases, eg some kind of long processing that takes on the order of minutes per record, it might be a very very long time before the consumer gets back to poll and logs the message you added. For the latter case, I think it would be valuable to keep this part about increasing the max.poll.interval or lowering the max.poll.records in the message we log here, when the max poll interval is first missed, so that users know what to do immediately and don't have to wait until they actually get through all 1000 records (or whatever max.poll.records is set to) and finally return to poll to see a hint about which configs to change

Reply:
Done, I did like the simplified log but totally agree with your points, both. I've been myself pushing for avoiding changing the existing logs content when possible because I've also heard about customers basing their apps on them. Also agree about the more complete output on the case of not hitting the next poll. 

So left the log here unchanged (and simplified the other just to not repeat ourselves on the 2 logs). So in the common case that we end up with the 2 log lines, it's just a first one about the situation when it happens, and the 2nd one with the approximate exceeded time when we have the most accurate info. Makes sense?